source/non_matching/coupling.cc

```diff
@@ -143,18 +148,44 @@ namespace NonMatching
       unsigned int             c = 0;
       for (const auto &cell : immersed_dh.active_cell_iterators())
         {
-          // Reinitialize the cell and the fe_values
-          fe_v.reinit(cell);
-          const std::vector<Point<spacedim>> &x_points =
-            fe_v.get_quadrature_points();
-
-          // Copy the points to the vector
-          std::copy(x_points.begin(),
-                    x_points.end(),
-                    all_points.begin() + c * n_q_points);
+          bool use_cell = true;
+          if (optimize_mpi)
+            {
+              const auto &bbox = cell->bounding_box();
+              std::vector<std::pair<
+                BoundingBox<spacedim>,
+                typename Triangulation<dim0, spacedim>::active_cell_iterator>>
+                out_vals;
+              cache.get_cell_bounding_boxes_rtree().query(
+                boost::geometry::index::intersects(bbox),
+                std::back_inserter(out_vals));
+              use_cell = false;
+              // Check if the cell overlaps a locally owned cell
```

Comment:
I don't really understand this comment. Should this be 'check if a bounding box overlaps a locally owned cell'?